store.go

```diff
@@ -1044,12 +1044,15 @@ func (s *store) SetNames(id string, names []string) error {
 	}
 
 	if rlstore.Exists(id) {
+		defer rlstore.Touch()
```

Comment:
Hmm, it's a pretty reasonable idea to move `Touch()` calls into the item store's `Save()` method, and out of the `Store`'s methods in general.